codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -168,30 +169,102 @@ private static void encode3to4(
         // significant bytes passed in the array.
         // We have to shift left 24 in order to flush out the 1's that appear
         // when Java treats a value as negative that is cast from a byte to an int.
-        int inBuff =
-                (numSigBytes > 0? src.getByte(srcOffset)     << 24 >>>  8 : 0) |
-                (numSigBytes > 1? src.getByte(srcOffset + 1) << 24 >>> 16 : 0) |
-                (numSigBytes > 2? src.getByte(srcOffset + 2) << 24 >>> 24 : 0);
+        if (dest.order() == ByteOrder.BIG_ENDIAN) {
+            final int inBuff;
+            switch (numSigBytes) {
+                case 1:
+                    inBuff = toInt(src.getByte(srcOffset));
+                    break;
+                case 2:
+                    inBuff = toInt(src.getShort(srcOffset));
+                    break;
+                default:
+                    inBuff = numSigBytes <= 0 ? 0 : toInt(src.getMedium(srcOffset));
+                    break;
+            }
+            encode3to4BigEndian(inBuff, numSigBytes, dest, destOffset, alphabet);
+        } else {
+            final int inBuff;
+            switch (numSigBytes) {
+                case 1:
+                    inBuff = toInt(src.getByte(srcOffset));
+                    break;
+                case 2:
+                    inBuff = toInt(src.getShortLE(srcOffset));
```

Comment:
we can reduce the operations if we do the byte swap and shifts in one operation:

```java
private static int toIntLE(short valueBE) {
  return (valueBE & 0xff) << 16 | (valueBE & 0xff00);
}
```